iree/compiler/Dialect/Flow/IR/test/tensor_folding.mlir

```diff
@@ -422,3 +422,14 @@ func @foldSplatLoadIntoPrimitive(%arg0 : f32, %arg1 : index, %arg2 : index) -> f
   %1 = flow.tensor.load %0[%arg1, %arg2] : tensor<4x4xf32>
   return %1 : f32
 }
+
+// -----
+
+// CHECK-LABEL: @foldSplatReshapeIntoSplat
+func @foldSplatReshapeIntoSplat(%arg0 : f32) -> tensor<16xf32> {
```

Comment:
would be worth making this test dynamic with different ssa values for a dim - that way you know that if it was `tensor<?x4xf32>{%foo}` reshaped to `tensor<?x?xf32>{%bar, %taco}` your new splat ends up with bar/taco. mm 🌮